packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Models/Types/SystemModelSerializationTypeProvider.cs

```diff
@@ -0,0 +1,231 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.ClientModel.Primitives;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.Json;
+using Microsoft.Generator.CSharp.Expressions;
+
+namespace Microsoft.Generator.CSharp.ClientModel
+{
+    /// <summary>
+    /// This class provides the set of serialization models, methods, and interfaces for a given model.
+    /// </summary>
+    internal sealed class SystemModelSerializationTypeProvider : TypeProvider
```

Comment:
Lets call this JsonMrwSerializationTypeProvider since that its the implementation of the json ModelReaderWriter